### PR TITLE
qemu-check: unhardcode ../sof.git and invoke from more places

### PR DESCRIPTION
**Pull request by marc-hb**
See commit messages.

To actually use the new flexibility https://github.com/thesofproject/qemu/pull/41 is also required.

### CODE REVIEW

File: scripts/qemu-check.sh [hunk 1]
```diff
@@ -119,27 +145,30 @@ do
...
 	if $has_rom; then
-		ROM="-r ../sof.git/build_${platform}_gcc"
+		ROM="-r ${SOF_BUILDS}/build_${platform}_gcc"
 		ROM+="/src/arch/xtensa/rom-$platform.bin"
 	fi
 
-	./xtensa-host.sh $PLATFORM -k \
-		../sof.git/build_${platform}_gcc/src/arch/xtensa/$FWNAME $ROM \
-		-o 2.0 ../sof.git/dump-$platform.txt
+        xtensa_host_sh=$(find_qemu_xtensa)
```
Comment (Bin-QA) [inline scripts/qemu-check.sh:152]: internal `die`, so it couldn't exit

File: scripts/qemu-check.sh [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
+usage: qemu-check.sh [ platform(s) ]
 	Supported platforms are ${SUPPORTED_PLATFORMS[*]}
+	Runs all supported platforms by default.
 EOF
 }
 
+find_qemu_xtensa()
+{
+	local xhs=xtensa-host.sh
+	for d in . ../qemu* qemu*; do
```
Comment (Bin-QA) [inline scripts/qemu-check.sh:34]: query for relative path?

File: scripts/qemu-check.sh (hunk 1 above)
Comment (marc-hb) [reply scripts/qemu-check.sh:152]: It exits thanks to `set -e`

File: scripts/qemu-check.sh (hunk 2 above)
Comment (marc-hb) [reply scripts/qemu-check.sh:34]: I don't understand sorry.

Comment (Bin-QA) [reply scripts/qemu-check.sh:34]: this part I'm not confirm, how about your current directory when run the script   
if current directory in sof, it is OK, if it is not ...

Comment (marc-hb) [reply scripts/qemu-check.sh:34]: This works in a limited number of directories, namely in `SOF/` and `SOF/*/`. Would like me to add a comment?

Comment (Bin-QA) [reply scripts/qemu-check.sh:34]: I think you don't need to add this comment

**Comment by dbaluta** [comment]
@marc-hb this mostly looks good to me! For future patches please try to separate patches so that 1 patch does one single thing!

This way, reviewing the code is easier and later on reading the git history would be easier to understand!

File: scripts/qemu-check.sh [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 # SPDX-License-Identifier: BSD-3-Clause
```
Comment (dbaluta) [inline scripts/qemu-check.sh:2]: Commit message says 'add set -e' the body of the message says that we add ignore for time out. This should really be two separate patches.

Comment (marc-hb) [reply scripts/qemu-check.sh:2]: The commit message is probably a bit too short, I meant: ignore set -e when calling qemu because qemu always times out.
